pkg/apis/kappctrl/v1alpha1/types_template.go

```diff
@@ -149,3 +150,15 @@ type AppTemplateCue struct {
 	// Cue expression to output, default will export all visible fields (optional)
 	OutputExpression string `json:"outputExpression,omitempty" protobuf:"bytes,4,opt,name=outputExpression"`
 }
+
+// +k8s:openapi-gen=true
+type AppTemplateOLMRegistry struct {
+	// BundleRoot is the path to bundle root within source directory.
+	// Default is the source directory root. (optional)
+	BundleRoot string `json:"bundleRoot,omitempty" protobuf:"bytes,1,opt,name=bundleRoot"`
+	// TargetNamespaces is the list of namespaces that the bundle will be configured to target.
+	// If the bundle supports AllNamespaces mode, default is all namespaces. Otherwise, if the
+	// bundle supports OwnNamespaces mode, default is the install namespace. If the bundle supports
+	// neither AllNamespaces nor OwnNamespace, TargetNamespaces must be specified.
+	TargetNamespaces []string `json:"targetNamespaces,omitempty" protobuf:"bytes,2,opt,name=targetNamespaces"`
```

Comment:
@joelanford what is the value of having TargetNamespaces here? Can't we get it directly from csv's install modes? In rukpak's implementation, we always pass an empty slice for target namespaces when converting from plain to registry: https://github.com/operator-framework/rukpak/blob/0e327c385ca5488da4c2eca95a68d385c7e08a5f/internal/convert/registryv1.go#L172 (and this was assuming we won't support multi-ns bundles). 

Are we looking to support multi namespace mode in v1, and accept that option from Operator API which would then be passed here?  

Follow up thought on that, if that's the case, we should handle it for the Helm and plain bundles also somehow. And probably App is the right place to expose it? Though we would not want App API to be showcased as "operator" specific.